src/Cache.ts

```diff
@@ -221,6 +219,28 @@ export class Cache {
             fileCache,
             file,
         );
+
+        // If there are no changes in any of the tasks, there's
+        // nothing to do, so just return.
+        if (Task.tasksListsIdentical(oldTasks, newTasks)) {
```

Comment:
I will probably move this line, and the creation of `oldTasks` above, in to a separate function, something like `tasksWereModifiedInFile(file.path, newTasks)...

This should make the logic easier to see, and the flow of code easier to read.

Reply:
Decided not to spend time on that now. The code is understandable. I can always pick this up if there are any further changes to this code...